src/net/gateway-archive.ts

```diff
@@ -0,0 +1,151 @@
+     *
+     * @param processId
+     * @param gateway
+     * @param errorMessage
+     */
+    export function getEnvelopeHeightArchive(processId: string, gateway: IGatewayClient, errorMessage: string) {
+        return getProcess(processId, gateway, errorMessage)
+            .then((response) => {
+                return {
+                    height: response.results.envelopeHeight
+                }
+            })
+    }
+
+    /**
+     * Fetches the archive results for a given processId
+     *
+     * @param processId
+     * @param gateway
+     * @param errorMessage
+     * @returns Results, vote process  type, vote process state
+     */
+    export function getRawResultsArchive(processId: string, gateway: IGatewayClient, errorMessage: string) {
+        return getProcess(processId, gateway, errorMessage)
+            .then((response) => {
+                return {
+                    results: response.results.votes,
+                    state: VochainProcessStatus[response.process.status as string] || "",
+                    height: response.results.envelopeHeight,
+                }
+            })
+    }
+
+    /**
+     * Fetch the process data on the IPFS archive
+     *
+     * @param processId
+     * @param gateway
+     * @param errorMessage
+     */
+    function getProcess(processId: string, gateway: IGatewayClient, errorMessage: string): Promise<IArchiveResponseBody> {
+        return resolveArchiveUri(gateway, errorMessage)
+            .then((archiveUri: ContentUri) => {
+                return getArchiveFile(archiveUri, processId, gateway)
+            })
+            .then((result: string) => JSON.parse(result))
+            .catch((error) => {
+                throw new GatewayArchiveError(errorMessage)
+            })
+    }
+
+    /**
+     * Resolves the archive Uri from the given network id
+     *
+     * @param gateway
+     * @param errorMessage
+     */
+    async function resolveArchiveUri(gateway: IGatewayClient, errorMessage: string): Promise<ContentUri> {
+        if (gateway.archiveIpnsId) {
+            return Promise.resolve(new ContentUri(gateway.archiveIpnsId))
+        }
+
+        const networkId = await gateway.networkId as EthNetworkID
+
+        return getEnsPublicResolverByNetwork(gateway, { environment: gateway.environment, networkId })
+            .then(ens => ens.instance.text(ens.entityEnsNode, TextRecordKeys.VOCDONI_ARCHIVE))
+            .then((uri: string) => {
+                if (!uri) {
+                    throw new GatewayArchiveError(errorMessage)
+                }
+                gateway.archiveIpnsId = uri
+                return new ContentUri(uri)
+            })
+    }
+
+    /**
+     * Gets the archive file from IPFS
+     *
+     * @param archiveUri
+     * @param processId
+     * @param gateway
+     */
+    function getArchiveFile(archiveUri: ContentUri, processId: string, gateway: IGatewayDVoteClient): Promise<string> {
```

Comment:
Could we name it `fetchArchivedProcess`?
Could it also return the parsed JSON data (+typedef) instead of leaving it to the caller?